kafka-client/src/main/java/dev/responsive/kafka/internal/utils/ExceptionSupplier.java

```diff
@@ -0,0 +1,36 @@
+package dev.responsive.kafka.internal.utils;
+
+import dev.responsive.kafka.internal.config.ResponsiveStreamsConfig;
+import java.util.Map;
+import org.apache.kafka.clients.consumer.CommitFailedException;
+import org.apache.kafka.common.errors.ProducerFencedException;
+import org.apache.kafka.streams.StreamsConfig;
+
+public class ExceptionSupplier {
```

Comment:
nit: rename to `StoreFencedExceptionSupplier` or something more specific

Reply:
ah nvm I saw you've named the function name to be more specific. So the idea is we'd have other types of exceptions supplied here as well? That's fine by me.